apps/token/src/routes/staking/staking-node.tsx

```diff
@@ -0,0 +1,149 @@
+import './staking-node.scss';
+
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+import { useParams } from 'react-router-dom';
+
+import { EpochCountdown } from '../../components/epoch-countdown';
+import { Colors } from '../../config';
+import type { VegaKeyExtended } from '@vegaprotocol/wallet';
+import { BigNumber } from '../../lib/bignumber';
+import type { Staking as StakingQueryResult } from './__generated__/Staking';
+import { ConnectToVega } from './connect-to-vega';
+// import { PendingStake } from "./pending-stake";
+import { StakingForm } from './staking-form';
+import { StakingNodesContainer } from './staking-nodes-container';
+import { StakingWalletsContainer } from './staking-wallets-container';
+import { ValidatorTable } from './validator-table';
+import { YourStake } from './your-stake';
+
+export const StakingNodeContainer = () => {
+  return (
+    <StakingWalletsContainer>
+      {({ currVegaKey }) =>
+        currVegaKey ? (
+          <StakingNodesContainer>
+            {({ data }) => <StakingNode vegaKey={currVegaKey} data={data} />}
+          </StakingNodesContainer>
+        ) : (
+          <ConnectToVega />
+        )
+      }
+    </StakingWalletsContainer>
+  );
+};
+
+interface StakingNodeProps {
+  vegaKey: VegaKeyExtended;
+  data?: StakingQueryResult;
+}
+
+export const StakingNode = ({ vegaKey, data }: StakingNodeProps) => {
+  const { node } = useParams<{ node: string }>();
+  const { t } = useTranslation();
+
+  const nodeInfo = React.useMemo(() => {
+    return data?.nodes?.find(({ id }) => id === node);
+  }, [node, data]);
+
+  const currentEpoch = React.useMemo(() => {
+    return data?.epoch.id;
+  }, [data?.epoch.id]);
+
+  const stakeThisEpoch = React.useMemo(() => {
+    const delegations = data?.party?.delegations || [];
+    const amountsThisEpoch = delegations
+      .filter((d) => d.node.id === node)
+      .filter((d) => d.epoch === Number(currentEpoch))
+      .map((d) => new BigNumber(d.amountFormatted));
+    return BigNumber.sum.apply(null, [new BigNumber(0), ...amountsThisEpoch]);
+  }, [data?.party?.delegations, node, currentEpoch]);
+
+  const stakeNextEpoch = React.useMemo(() => {
+    const delegations = data?.party?.delegations || [];
+    const amountsNextEpoch = delegations
+      .filter((d) => d.node.id === node)
+      .filter((d) => d.epoch === Number(currentEpoch) + 1)
+      .map((d) => new BigNumber(d.amountFormatted));
+
+    if (!amountsNextEpoch.length) {
+      return stakeThisEpoch;
+    }
+    return BigNumber.sum.apply(null, [new BigNumber(0), ...amountsNextEpoch]);
+  }, [currentEpoch, data?.party?.delegations, node, stakeThisEpoch]);
+
+  // const pendingStakeNextEpoch = React.useMemo(() => {
+  //   return stakeNextEpoch.minus(stakeThisEpoch);
+  // }, [stakeThisEpoch, stakeNextEpoch]);
```

Comment:
?

Reply:
This also just came over in the migration.

Reply:
Deleted